shaders/src/main.fs

```diff
@@ -40,7 +40,7 @@ void main() {
 #endif
 
 #if defined(HAS_FOG)
-    vec3 view = getWorldPosition() - getWorldCameraPosition();
+    vec3 view = getWorldCameraForward();
```

Comment:
Note that `view` should not be a normalized value. This change is not only incorrect in ortho, it breaks perspective. Instead, `view` should be
- in perspective: kept as is, as the vector from eye to fragment position,
- in ortho: the vector from the fragments's orthogonal projection on the near plane to fragment position.

Also `light_punctual.fs` should receive the same treatment.

Reply:
Nevermind about `light_punctual.fs`, I was on a newer upstream main ([light_punctual.fs#L110](https://github.com/google/filament/blob/afd657d6af0dc7ead4a083685a6e13815e0a9902/shaders/src/light_punctual.fs#L110)), it will have to wait until we catch up with upstream.

Reply:
Good catch. My idea for detecting such pitfalls is to rename `getWorldCameraPosition()` (e12ac5e85118f0f266f4f52161ee6720ae4b6b26). This way `light_punctual.fs` will fail to compile when we bump to a newer upstream, forcing us to review it.

Reply:
Eventually I got rid of `getWorldCameraPosition()` because it was being used incorrectly in Filament (see 63cc63b0b57a3f7c15005351487835f82cd2972e). I updated the usages. In future bumps, `light_punctual.fs` will fail to compile, letting us know it needs our attention.